src/types.cpp

```diff
@@ -1108,29 +1120,26 @@ stdx::optional<wf::output_config::mode_t> wf::option_type::from_string(
         return wf::output_config::mode_t{true};
     }
 
+    std::string resolution,from;
+
     if (string.substr(0, 6) == "mirror")
     {
         std::stringstream ss(string);
-        std::string from, dummy;
         ss >> from; // the mirror word
-        if (!(ss >> from))
+        if (ss >> from)
```

Comment:
I am confused, is the code not supposed to continue running after that, so that we can parse the mode as well?

Also, I think it may be simpler (and more flexible wrt. whitespace handling) to try sscanf multiple times. One time with a format string like `mirror %s <optionally the mode>`, and if that fails, then scan a normal resolution, like the code below does.

Reply:
[59-1.txt](https://github.com/WayfireWM/wf-config/files/15247051/59-1.txt)
I've modified the logic a bit so the program execution is not prematurely finished before the mode string parsed.  So now it works - but the code is a bit ugly. I am not C++ expert ufortunately. But for personal use that works.